enhancements/cluster-scope-secret-volumes/shared-resource-validation.md

```diff
@@ -0,0 +1,413 @@
+in a ValidatingAdmissionWebhook could be impactful, as we are ultimately gating whether a Pod is allowed or not.  If 
+there is a bug which results in valid Pods being rejected, that could have broad and major consequences.
+
+Does that risk give enough pause so that we punt on the user story around more useful failure modes when the user has 
+failed to set 'readOnly: true' on their Pod's CSI Volume mount?  The fact that there is general agreement on pursuing 
+this EP, the assumption is the answer to that question is "no, we'll just be careful to get it right".  But let's 
+discuss as needed.
+
+The OCP teams possibly involved with issues:
+- API Server (webhook invocation, confusion around creating/modifying objects)
+- Kubelet/node (incorrectly perhaps if valid Pods fail to come up, unless there is a problem beyond this EP's validations)
+- Cluster Storage Operator (currently storage team, if a CSO problem leads to the webhook not coming up) 
+- Shared Resource Operator/Driver themselves (currently build/jenkins team)
+
+
+#### Support Procedures
+
+##### Detect Failure modes and Possible Symptoms
+
+- If the webhook is not running, kube-apiserver logs will show errors like "failed to call admission webhook xyz".
+- Cluster Storage Operator / CSI Shared Resource Operator will degrade with message "Failed to launch webhook server" and reason "WehhookServerFailed".
+- The k8s metric `webhook_admission_duration_seconds("<name of our new admission controller>", "validating", "put", "false")`
+    will show >1s latency and alert `WebhookAdmissionLatencyHigh` will fire.
+- Otherwise, we should be able to fall back on must gather.
+
+##### Disable the API extension (e.g. remove ValidatingWebhookConfiguration `<names of the our new webhook configs>`)
+
+- What consequences does it have on the cluster health?
+
+There are no consequences on cluster health for disabling the new ValidatingAdmissionWebhook(s).  They provide
+for better user experience with respect to error reporting when incorrect settings are provided Shared Resource instances
+or Pods (and their descendents) that specific CSI volumes using the Shared Resource CSI Driver.  But the controller / CSI Driver
+can also catch those errors, though as the Goals/User Stories note, the user has to do more digging.
+      
+
+- What consequences does it have on existing, running workloads?
+
+There are no consequences on existing, running workloads.  Only the Shared Resource Controller and CSI Driver impact
+existing, running workloads that have Shared Resource CSI Volume mounts.
+
+- What consequences does it have for newly created workloads?
+
+For properly constructed Shared Resources and Pods with Shared Resource CSI Volume Mounts, incorrect specifications
+will still be denied, but the errors will not manifest on Pod creation, but when Pod containers are trying to start.
+
+## Implementation History
+
+Initial submission of this EP
+
+## Drawbacks
```

Comment:
Adding a webhook that watches all pod creation risks cluster stability.

Reply:
I mentioned that concern earlier but sure I can mention it here, though per the other thread, maybe we switch on that for the Pod validations.